enetwork

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+
+## Variables, fixed for now, will be env vars or secrets later
+GatewayUsername=${SEMAPHORE_DOCKERGW_USER,-''}
+GatewayPassword=${SEMAPHORE_DOCKERGW_PASSWORD,-''}
+GatewayEndPoint=${SEMAPHORE_DOCKERGW_ENDPOINT,-''}
+GatewayEndPointPort=${SEMAPHORE_DOCKERGW_ENDPOINT_PORT,-''}
+
+## Create docker env change directory
+sudo mkdir -p /etc/systemd/system/docker.service.d
+
+## Create proxy.conf
+en_start(){
+
+  CanSetProxy=$(curl --write-out %{http_code} --head --silent --output /dev/null -x http://"$GatewayUsername":"$GatewayPassword"@"$GatewayEndPoint":"$GatewayEndPointPort" https://www.google.com)
+
+  if [ "$CanSetProxy" -eq "200" ]; then
+    echo "[Service]"                                                                                                       | sudo tee    /etc/systemd/system/docker.service.d/proxy.conf > /dev/null
+    echo 'Environment="HTTP_PROXY=http://'"$GatewayUsername"':'"$GatewayPassword"'@'"$GatewayEndPoint"':'"$GatewayEndPointPort"'"' | sudo tee -a /etc/systemd/system/docker.service.d/proxy.conf > /dev/null
+    echo 'Environment="HTTPS_PROXY=http://'"$GatewayUsername"':'"$GatewayPassword"'@'"$GatewayEndPoint"':'"$GatewayEndPointPort"'"'| sudo tee -a /etc/systemd/system/docker.service.d/proxy.conf > /dev/null
+    echo "Environment=\"NO_PROXY='*.semreg.com,registry.semaphoreci.com,localhost,127.0.0.1,::1'\""                        | sudo tee -a /etc/systemd/system/docker.service.d/proxy.conf > /dev/null
+    echo 'Docker gateway is set, reloading docker daemon'
+    en_reload && echo 'Docker daemon reloaded' || echo 'Failed to reload docker daemon'
+  else
+    echo 'Authentication error, not setting docker gateway'
+  fi
+}
```

Comment:
The code looks good, but it would be wise to improve the readability/maintainability of the code.

Here are my proposals:

**suggestion 1**
Extract the heavy part of the code to a dedicated function. This way you won't have to really a comment on the code to communicate what it does, it will be obvious from the function name:

```
en_create_proxy_conf() {
    echo "[Service]"                                                                                                       | sudo tee    /etc/systemd/system/docker.service.d/proxy.conf > /dev/null
    echo 'Environment="HTTP_PROXY=http://'"$GatewayUsername"':'"$GatewayPassword"'@'"$GatewayEndPoint"':'"$GatewayEndPointPort"'"' | sudo tee -a /etc/systemd/system/docker.service.d/proxy.conf > /dev/null
    echo 'Environment="HTTPS_PROXY=http://'"$GatewayUsername"':'"$GatewayPassword"'@'"$GatewayEndPoint"':'"$GatewayEndPointPort"'"'| sudo tee -a /etc/systemd/system/docker.service.d/proxy.conf > /dev/null
    echo "Environment=\"NO_PROXY='*.semreg.com,registry.semaphoreci.com,localhost,127.0.0.1,::1'\""                        | sudo tee -a /etc/systemd/system/docker.service.d/proxy.conf > /dev/null  
}
```

**suggestion 2**
Be consistent with spacing in your file. The first thing I noticed while opening this file is that you are inconsistent with spacing and cleanness in your code. This significantly detracts from our ability to review the code, as we start fixating on improper syntax and ignore the meat of the code:

Do this:

```
fun_a() {
  code
  code
  code
}

fun_b() {
  code
  code
  code
}

fun_c() {
  code
  code
}
```

Instead of this:

```
fun_a(){

  code
  code
}
fun_b() {
  code
  code
}


fun_c() {
  code
  code
}
```

Reply:
When in doubt, this is a great guide to follow https://google.github.io/styleguide/shellguide.html.

Reply:
You could simplify it, but using just one command:

```
sudo tee -a /etc/systemd/system/docker.service.d/proxy.conf > /dev/null <<EOT
[Service]
Environment="HTTP_PROXY=http://$GatewayUsername:$GatewayPassword@$GatewayEndPoint:$GatewayEndPointPort"
Environment="HTTPS_PROXY=http://$GatewayUsername:$GatewayPassword@$GatewayEndPoint:$GatewayEndPointPort"
Environment="NO_PROXY='*.semreg.com,registry.semaphoreci.com,localhost,127.0.0.1,::1'"
EOT
```

Reply:
Done.